torch/_refs/linalg/__init__.py

```diff
@@ -57,6 +58,16 @@ def check_norm_dtype(dtype: Optional[torch.dtype], x_dtype: torch.dtype, fn_name
 from torch._decomp import register_decomposition
 
 
+def diagonal(
+    input: TensorLikeType,
+    *,
+    offset: int = 0,
+    dim1: int = -2,
```

Comment:
cc'ing @nkaretnikov, who originally added this ref in #84792. I added this while working on #85419. Hope it is fine. :) 
 Thanks!